data/trie/pruningBuffer.go

```diff
@@ -0,0 +1,53 @@
+package trie
+
+import (
+	"bytes"
+	"sync"
+)
+
+type pruningBuffer struct {
+	mutOp  sync.RWMutex
+	buffer [][]byte
+}
+
+func newPruningBuffer() *pruningBuffer {
+	return &pruningBuffer{
+		buffer: make([][]byte, 0),
+	}
+}
+
+func (sb *pruningBuffer) add(rootHash []byte) {
+	sb.mutOp.Lock()
+	defer sb.mutOp.Unlock()
+
+	if len(sb.buffer) == pruningBufferLen {
+		log.Trace("pruning buffer is full", "rootHash", rootHash)
+		return
+	}
+
+	sb.buffer = append(sb.buffer, rootHash)
+	log.Trace("pruning buffer add", "rootHash", rootHash)
+}
+
+func (sb *pruningBuffer) remove(rootHash []byte) {
+	sb.mutOp.Lock()
+	defer sb.mutOp.Unlock()
+
+	for i := 0; i < len(sb.buffer); i++ {
```

Comment:
instead of delete from existing buffer, create a new one without those being equal with rootHash